### DISCUSSION TITLE
Client SDKs

### DISCUSSION THREAD

**Discussion by abelanger5** [hatchet-dev/hatchet#436] [Polls]
Vote for the next SDK that we should build out! And feel free to suggest any additional SDK options in the discussion and we'll add it to the poll.

**Comment by abelanger5** [comment]
**Elixir**

(can't update without resetting votes, so give this comment a thumbs-up for elixir)

**Comment by tracker1** [comment]
Kind of wish this was multiple choice... I put Rust as my vote, but would also like to see a C#/.Net library as well.

Of course each driver also means integrating with a different repository ecosystem.

**Comment by jakubadamw** [comment]
I'm taking a stab at building a simple [a Rust SDK](https://github.com/jakubadamw/hatchet-rust-sdk). I would like to use Hatchet for a side project of mine, and the project uses Rust for the workflows that I would like to move to Hatchet, so I figured I would go ahead and build a SDK. It's nowhere near close to being usable. I will post again when I get it working.

**Comment by abelanger5** [comment]
Hey @jakubadamw that's awesome to see! We'd love to give this a try/link to it from our documentation when this is working.

**Comment by musjj** [comment]
Were there any progress here? Would love to see a Rust SDK!

**Comment by PerfectedApp** [comment]
Maybe there's documentation on how to build a community version? Would love to play around with ziglang as a side project.

**Comment by eswolinsky3241** [comment]
Hey @tracker1  @musjj and @abelanger5. I've just published [version 0.1.0](https://github.com/eswolinsky3241/hatchet-rust-sdk/releases/tag/v0.1.0) of my Hatchet Rust SDK. If you have a chance to check it out, would appreciate any feedback!

**Comment by loleek** [comment]
Any plan for java sdk, please?

**Reply by caocao-sys** [reply]
We need this too

**Reply by Hellothi11** [reply]
We need this too

**Comment by tracker1** [comment]
@eswolinsky3241 sorry, I haven't looked at your solution... are you exposing a C-style dylib exports in your rust implementation?  Or is it strictly a rust ecosystem library?

Only asking as it could be a path towards C#/.Net or Java/JNI via C style library usage, at least for an initial pass, short of a full in-language implementation... that said, at this point, with clear API, a lot of Agents can probably do a lot of clone/port work relatively easily if someone wanted to steer/validate such an effort.

Not volunteering, just suggestion... I'm not using Hatchet currently, so I'd be a bad person for such work as a result.
